buildpacks/dotnet/src/dotnet/global_json.rs

```diff
@@ -30,22 +30,41 @@ impl TryFrom<SdkConfig> for VersionReq {
 
     // TODO: Factor in pre-release logic
     fn try_from(sdk_config: SdkConfig) -> Result<Self, Self::Error> {
-        let version = &sdk_config.version;
-        let roll_forward = sdk_config.roll_forward.as_deref();
-
-        let version_req_str = match roll_forward {
-            Some("patch" | "latestPatch") => format!("~{version}"),
-            Some("feature" | "latestFeature") => format!(
-                "~{}",
-                version.split('.').take(2).collect::<Vec<_>>().join(".")
-            ),
-            Some("minor" | "latestMinor") => format!(
-                "^{}",
-                version.split('.').take(2).collect::<Vec<_>>().join(".")
-            ),
-            Some("major" | "latestMajor") => "*".to_string(),
-            Some("disable") => format!("={version}"),
-            _ => version.clone(),
+        let version_str = sdk_config.version.as_ref();
+        // Parse version to ensure we have valid components to work with
+        let version = Version::parse(version_str)?;
+
+        // Default policy is `patch`, see https://learn.microsoft.com/en-us/dotnet/core/tools/global-json#matching-rules
+        let policy = sdk_config.roll_forward.as_deref().unwrap_or("patch");
+
+        let version_req_str = match policy {
+            "patch" | "latestPatch" => {
+                // Feature band logic: 6.0.1xx matches 6.0.1xx, but not 6.0.2xx.
+                // See https://learn.microsoft.com/en-us/dotnet/core/tools/global-json#rollforward
+                let patch = version.patch;
+                let feature_band_start = (patch / 100) * 100;
+                let feature_band_end = feature_band_start + 100;
+
+                // If the user requested a pre-release (e.g., 6.0.100-rc.1),
+                // we must allow pre-releases in the lower bound.
+                // Using an exact comparator for the lower bound handles this best.
+                format!(
+                    ">={}, <{}.{}.{}",
+                    version_str, // Use full string (6.0.100-rc.1) to include pre-release
+                    version.major,
+                    version.minor,
+                    feature_band_end
+                )
+            }
+            "feature" | "latestFeature" => {
+                format!("~{}.{}", version.major, version.minor)
+            }
+            "minor" | "latestMinor" => {
+                format!("^{}.{}", version.major, version.minor)
+            }
+            "major" | "latestMajor" => "*".to_string(),
+            "disable" => format!("={version_str}"),
+            _ => version_str.to_string(),
```

Comment:
(for the future) Should the buildpack warn or error if an invalid policy is specified?